tests/testthat/test-with_parameters.R

```diff
@@ -217,4 +217,20 @@ test_that("glue-formatted descriptions and test names supported", {
     ),
     "produced output of length 10"
   )
+
+  # but fail kindly for potential accidental use of glue
+  #   c.f. https://github.com/r-lib/lintr/issues/2706
+  expect_error(
+    with_parameters_test_that("a{b}", {
+      expect_true(TRUE)
+    }, .cases = data.frame(d = 1)),
+    "Attempt to interpret test stub 'a{b}' with glue failed",
+    fixed = TRUE
+  )
+  # as well as an escape hatch to work around needing ugly escapes
+  expect_success(
```

Comment:
@MichaelChirico Sorry, one last error is showing up in CI. `expect_success()` is for expectations. 
https://testthat.r-lib.org/reference/expect_success.html

We could instead:

*  Check for no errors (`check_no_error()`)
*  Or suppress the reporter and check the return value of the test: https://github.com/r-lib/testthat/blob/50a99ec76fed8140a9211ed862289e877e86614d/tests/testthat/test-test-that.R#L64

Reply:
I think `expect_no_error()` is good, given that it throws an error on `main` and we aren't really concerned here about interaction with {testthat}, but really about the code working at all. In other words, the error in the bug is present independently of {testthat}, so testing `expect_{success,failure}` is not exactly right IMO.